lib/shared/bucketing/src/bucketing.ts

```diff
@@ -326,3 +331,26 @@ export const generateBucketedConfig = ({
         variables: variableMap,
     }
 }
+
+export const getUserValueForBucketingKey = ({
+    user,
+    target,
+}: {
+    user: DVCBucketingUser
+    target: PublicTarget
+}): string => {
+    if (target.bucketingKey && target.bucketingKey !== 'user_id') {
+        const bucketingValue =
+            user.customData?.[target.bucketingKey] || DEFAULT_BUCKETING_VALUE
```

Comment:
this should also check against the `user.privateCustomData`

Reply:
seconding this, the `privateCustomData` might already have been merged with `customData` at this point though